### ISSUE TITLE
pylint-protobuf crash with 'ClassDef' object has no attribute 'doc'

### ISSUE THREAD

**Issue by matejsp** [nelfin/pylint-protobuf#58]
I have run the released version on our CI with pylint 3.0.1 and astroid 3.0.
AttributeError: 'ClassDef' object has no attribute 'doc'.
I cannot reproduce it locally just on CI where we run it on all code.

```
Creating folders
Creating files
grep: warning: stray \ before white space
Found 0 pylint warnings on HEAD branch
Running pylint mode with zero pylint errors tolerance enabled
Exception on node <ImportFrom l.10 at 0x7f1e82e****bd0> in file '/bitstamp/common/audit_log/audit_log.py'
Traceback (most recent call last):
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 2055, in visit_importfrom
    self._check_module_attrs(node, module, name.split("."))
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 3057, in _check_module_attrs
    module = next(module.getattr(name)[0].infer())
                  ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 353, in getattr
    result = [self.import_module(name, relative_only=True)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 453, in import_module
    return Astroid****anager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 259, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 151, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line ****0, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line 168, in _post_build
   module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 1****, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 151, in visit
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 84, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 66, in _transform
    ret = transform_func(node)
          ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 578, in transform_module
    for local_name, node in transform_descriptor_to_class(cls):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 528, in transform_descriptor_to_class
    return transform_enum(desc, desc_registry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 328, in transform_enum
    simple_desc = descriptor_registry[cls_def.doc.split('=')[-1]]  # FIX****E: guard?
                                      ^^^^^^^^^^^
AttributeError: 'ClassDef' object has no attribute 'doc'
Exception on node <ImportFrom l.6 at 0x7f1e75ba2e50> in file '/bitstamp/common/audit_log/grpc/user_history_client.py'
Traceback (most recent call last):
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 2055, in visit_importfrom
    self._check_module_attrs(node, module, name.split("."))
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 3057, in _check_module_attrs
    module = next(module.getattr(name)[0].infer())
                  ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 353, in getattr
    result = [self.import_module(name, relative_only=True)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 453, in import_module
    return Astroid****anager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 259, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 151, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line ****0, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line 168, in _post_build
    module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 1****, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 151, in visit
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 84, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 66, in _transform
    ret = transform_func(node)
          ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 578, in transform_module
    for local_name, node in transform_descriptor_to_class(cls):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 528, in transform_descriptor_to_class
    return transform_enum(desc, desc_registry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 328, in transform_enum
    simple_desc = descriptor_registry[cls_def.doc.split('=')[-1]]  # FIX****E: guard?
                                      ^^^^^^^^^^^
AttributeError: 'ClassDef' object has no attribute 'doc'
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 831, in _check_file
    check_astroid_module(ast_node)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 1017, in check_astroid_module
    retval = self._check_astroid_module(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 1069, in _check_astroid_module
    walker.walk(node)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 2055, in visit_importfrom
    self._check_module_attrs(node, module, name.split("."))
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/checkers/variables.py", line 3057, in _check_module_attrs
    module = next(module.getattr(name)[0].infer())
                  ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 353, in getattr
    result = [self.import_module(name, relative_only=True)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 453, in import_module
    return Astroid****anager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 259, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 151, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line ****0, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/builder.py", line 168, in _post_build
    module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/manager.py", line 1****, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 151, in visit
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 84, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/astroid/transforms.py", line 66, in _transform
    ret = transform_func(node)
          ^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 578, in transform_module
    for local_name, node in transform_descriptor_to_class(cls):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 528, in transform_descriptor_to_class
    return transform_enum(desc, desc_registry)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint_protobuf/transform.py", line 328, in transform_enum
    simple_desc = descriptor_registry[cls_def.doc.split('=')[-1]]  # FIX****E: guard?
                                      ^^^^^^^^^^^
AttributeError: 'ClassDef' object has no attribute 'doc'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 256, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 205, in _process_chunk
    return [fn(*args) for args in chunk]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 205, in <listcomp>
    return [fn(*args) for args in chunk]
            ^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/parallel.py", line 79, in _worker_check_single_file
    _worker_linter.check_single_file_item(file_item)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 740, in check_single_file_item
    self._check_file(self.get_ast, check_astroid_module, file)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 833, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/virtualenvs/py311/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
             ^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/__init__.py", line 34, in run_pylint
    PylintRun(argv or sys.argv[1:])
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/run.py", line 211, in __init__
    linter.check(args)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 679, in check
    check_parallel(
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/parallel.py", line 153, in check_parallel
    for (
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 597, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 619, in result_iterator
    yield _result_or_cancel(fs.pop())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 317, in _result_or_cancel
    return fut.result(timeout)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
astroid.exceptions.AstroidError
Pylint should not crash while checking code, please fix incompatibilities.

```

**Comment by matejsp** [comment]
Remove deprecated doc attribute for Module, ClassDef, and FunctionDef. Use the doc_node attribute instead.
Refs #2154

https://pylint.readthedocs.io/projects/astroid/en/latest/changelog.html#what-s-new-in-astroid-3-0-0

**Comment by matejsp** [comment]
And another crash with pylint 3.0 (after fixing the previous)

```
Traceback (most recent call last):
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/project/pylint_protobuf/__init__.py", line 177, in visit_call
    self._check_init_kwargs(node)
  File "/project/pylint_protobuf/__init__.py", line 215, in _check_init_kwargs
    self._check_kwargs(desc, node)
  File "/project/pylint_protobuf/__init__.py", line 241, in _check_kwargs
    desc_name = desc.name
                ^^^^^^^^^
AttributeError: 'list' object has no attribute 'name'
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 831, in _check_file
    check_astroid_module(ast_node)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 1017, in check_astroid_module
    retval = self._check_astroid_module(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 1069, in _check_astroid_module
    walker.walk(node)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
 File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  [Previous line repeated 3 more times]
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/project/pylint_protobuf/__init__.py", line 177, in visit_call
    self._check_init_kwargs(node)
  File "/project/pylint_protobuf/__init__.py", line 215, in _check_init_kwargs
    self._check_kwargs(desc, node)
  File "/project/pylint_protobuf/__init__.py", line 241, in _check_kwargs
    desc_name = desc.name
                ^^^^^^^^^
AttributeError: 'list' object has no attribute 'name'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 256, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 205, in _process_chunk
    return [fn(*args) for args in chunk]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 205, in <listcomp>
    return [fn(*args) for args in chunk]
            ^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/parallel.py", line 79, in _worker_check_single_file
    _worker_linter.check_single_file_item(file_item)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 740, in check_single_file_item
    self._check_file(self.get_ast, check_astroid_module, file)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 833, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/virtualenvs/py311/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
             ^^^^^^^^^^^^
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/__init__.py", line 34, in run_pylint
    PylintRun(argv or sys.argv[1:])
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/run.py", line 211, in __init__
    linter.check(args)
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/pylinter.py", line 679, in check
    check_parallel(
  File "/opt/virtualenvs/py311/lib64/python3.11/site-packages/pylint/lint/parallel.py", line 153, in check_parallel
    for (
  File "/usr/lib64/python3.11/concurrent/futures/process.py", line 597, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 619, in result_iterator
   yield _result_or_cancel(fs.pop())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 317, in _result_or_cancel
    return fut.result(timeout)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
astroid.exceptions.AstroidError
Pylint should not crash while checking code, please fix incompatibilities.
```

**Comment by matejsp** [comment]
@nelfin Can you check my pr if it is ok and prepare maybe new release that fixed pylint 3.x ? #59

**Comment by nelfin** [comment]
Thank you for the pull request. For your awareness, I won't have time to
address it until this weekend. The changes to astroid 3.x also broke some
utilities that mean the unit tests no longer run, which hampers my ability
to quickly verify that no other regressions are introduced.

**Comment by matejsp** [comment]
@vasanthzoh and others. While waiting for a proper release you can test my PR:
```
pip install git+https://github.com/matejsp/pylint-protobuf.git@c85cafe998020274edd73fbcbfad818a1fce03ad  
```
It works with our code without any problem.

**Comment by matejsp** [comment]
@nelfin sry to bother you, but any chance to merge the PR and release new version? :)

**Comment by nelfin** [comment]
Thanks for your patience, after updating the test suite and a few more fixes I've released 0.22 with support for astroid 2.9.x to 3.0.x

**Comment by matejsp** [comment]
It works with latest release. Great job!
